### DISCUSSION TITLE
Table组件能不能加个自定义字段Key原封不动的带进QueryPageOptions就可以

### DISCUSSION THREAD

**Discussion by shur520** [dotnetcore/BootstrapBlazor#3737] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

用DataServiceBase的QueryAsync统一查询数据的时候没办法知道这个查询来自哪个Table
加个参数可以做个识别,也能做权限判断,不然每个QueryAsync都要写到页面里.

### Describe the solution you'd like

加个自定义字段串就行,在QueryPageOptions能获取到就行

**Comment by bb-auto** [comment]
@shur520 Thank you for contacting us. We will give feedback later.

**Comment by j4587698** [comment]
服务不应该跟UI有关系，服务就是服务，只应该接收参数，返回响应

**Reply by ArgoZhang** [reply]
@shur520 设计上数据服务仅提供满足条件的数据，不需要知道谁再请求，请求数据服务的也不一定是 `Table` 组件，大佬是有什么需求？可以说一下看看有没有别的方案。

**Reply by shur520** [reply]
我想在Table设置一个数据范围的权限代码,查询数据的时候根据这个代码的权限范围返回相应的数据,比例这个代码对应的是本部门数据范围,就只返回本部门数据

你可能觉得获取当前用户角色的数据范围就可以,但是我的系统一个角色有N多个数据范围,是根据菜单设置范围的,需要知道他访问的是哪个列表来决定数据范围

**Reply by j4587698** [reply]
> 我想在Table设置一个数据范围的权限代码,查询数据的时候根据这个代码的权限范围返回相应的数据,比例这个代码对应的是本部门数据范围,就只返回本部门数据
> 
> 你可能觉得获取当前用户角色的数据范围就可以,但是我的系统一个角色有N多个数据范围,是根据菜单设置范围的,需要知道他访问的是哪个列表来决定数据范围


估计你本来就自己包了一层Table，只需要把你自己包的那个Table稍微处理一下就行了，然后把DataService自己加个参数，就搞定了。

**Reply by ArgoZhang** [reply]
@shur520 `Table` 内置一个搜索条件 `权限代码` 数据服务那边检查这个 `权限代码` 给数据，比如网页1 的第一个表格 权限代码就是 `101`

**Comment by densen2014** [comment]
自己实现一个服务, 注入你的页面条件就行了
